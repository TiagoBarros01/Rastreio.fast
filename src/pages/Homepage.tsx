import React, { useContext, useState } from 'react';
import { ThemeProvider } from 'styled-components';

import Footer from '../components/Footer';
import { TrackingContext } from '../contexts/TrackingContext';
import { Input, Title, TrackBoxContainer } from '../styles/pages/Homepage';
import dark from '../styles/themes/dark';

export default function Homepage() {
  const [code, setCode] = useState<string>('');

  const {
    getTrackingData,
  } = useContext(TrackingContext);

  return (
    <ThemeProvider theme={dark}>
      <TrackBoxContainer>
        <Title>
          <h1>
            Rastreio
            <span>.</span>
            fast
          </h1>
        </Title>
        <div>
          <Input
            maxLength="15"
            type="text"
            value={code.toUpperCase()}
            onChange={(event) => setCode(event.target.value)}
            autoComplete="on"
            placeholder="Code here"
            required
          />
          <button
            type="button"
            onClick={() => getTrackingData(code)}
          >
            Track order
          </button>
        </div>
      </TrackBoxContainer>
      <Footer />
    </ThemeProvider>
  );
}
