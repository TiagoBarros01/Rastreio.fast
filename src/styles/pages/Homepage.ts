import styled from 'styled-components';

export const TrackBoxContainer = styled.div`
  width: 100vw;
  height: 100vh;

  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  button {
    padding: 1rem;

    border: 0;

    background: ${({ theme }) => theme.colors.thirdBackground};

    transition: background-color 200ms;

    @media (max-width: 576px) {
      margin-top: 1rem;
      width: 45%;
    }

    :hover {
      background: ${({ theme }) => theme.colors.yellow};
    }
  }
`;

export const CodeContainer = styled.div`
  @media (max-width: 576px) {
    display: flex;

    justify-content: center;
    align-items: center;
    flex-direction: column;
  }
`;

export const Title = styled.div`
  width: 100%;
  margin: 0 auto;

  text-align: center;

  margin-bottom: 1rem;

  h1 {
    color: ${({ theme }) => theme.colors.text};
    font-size: 3rem;

    span {
      color: ${({ theme }) => theme.colors.yellow};
    }
  }
`;

export const Input = styled.input`
  width: 20rem;
  padding: 1rem;

  background: ${({ theme }) => theme.colors.inputBackground};
  color: ${({ theme }) => theme.colors.inputText};

  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;

  ::placeholder {
    color: ${({ theme }) => theme.colors.inputText};
  }
`;
