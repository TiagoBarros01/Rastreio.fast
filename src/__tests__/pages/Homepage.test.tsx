import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import ContextProvider from '../../contexts';
import Homepage from '../../pages/Homepage';

describe('Test Homepage successfully', () => {
  it('Should homepage exists', () => {
    const { container } = render(
      <BrowserRouter>
        <ContextProvider>
          <Homepage />
        </ContextProvider>
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should button exists', () => {
    const btnText = /track order/i;

    render(
      <BrowserRouter>
        <ContextProvider>
          <Homepage />
        </ContextProvider>
      </BrowserRouter>,
    );

    screen.getByRole('button', { name: btnText });
  });

  it('Should button clicked', () => {
    const btnText = /track order/i;

    render(
      <BrowserRouter>
        <ContextProvider>
          <Homepage />
        </ContextProvider>
      </BrowserRouter>,
    );

    const button = screen.getByRole('button', { name: btnText });

    userEvent.click(button);
  });

  it('Should input exists', () => {
    render(
      <BrowserRouter>
        <ContextProvider>
          <Homepage />
        </ContextProvider>
      </BrowserRouter>,
    );

    screen.getByPlaceholderText(/Code here/i);
  });
});
