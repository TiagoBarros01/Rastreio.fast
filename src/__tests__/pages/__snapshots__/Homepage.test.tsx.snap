// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Homepage successfully Should homepage exists 1`] = `
<div>
  <div
    class="sc-bqiQRQ kgKXYI"
  >
    <div
      class="sc-hBURRC gKDKkA"
    >
      <h1>
        Rastreio
        <span>
          .
        </span>
        fast
      </h1>
    </div>
    <div
      class="sc-ksdxAp gdNJwt"
    >
      <input
        class="sc-fotPbf fnQypi"
        maxlength="15"
        placeholder="Code here"
        required=""
        type="text"
      />
      <button
        type="button"
      >
        Track order
      </button>
    </div>
  </div>
  <div
    class="sc-egiSv fFrwtl"
  >
    <p>
      Created with ðŸ’› by
       
      <a
        href="/https://github.com/TiagoBarros01"
        rel="noreferrer"
        target="_blank"
      >
        Tiago Barros
      </a>
    </p>
  </div>
</div>
`;
